packages/pinia/src/store.ts

```diff
@@ -880,6 +880,14 @@ export function defineStore(
     id = idOrOptions.id
   }
 
+  // Warn developers about existing store ID in dev environments
+  if (__DEV__ && activePinia?._k.includes(id)) {
+    console.warn(`[🍍]: Stores should have unique identifiers. Found multiple stores with id "${id}". Rename one of them.`)
+  }
+
+  // Push ID to keys registry
+  activePinia?._k.push(id)
```

Comment:
I think you only need to push the key when `__DEV__` is true. There's no need to push it in production, as this will help reduce memory consumption